File: modules/wem-webapp/src/main/webapp/admin/common/js/expr/OrderExpr.ts
module api.expr {

    export enum Direction {
        ASC,
        DESC
    }

    export class OrderExpr implements Expression {

        private direction:Direction;

        constructor( direction:Direction )
        {
            this.direction = direction;
        }

        getDirection():Direction {
            return this.direction;
        }

        directionAsString():string {
            switch (this.direction) {
                case Direction.ASC:
                    return "ASC";
                case Direction.DESC:
                    return "DESC";
                default:
                    return "";
            }
        }
    }
}

File: modules/wem-api/src/main/java/com/enonic/wem/api/query/expr/OrderExpr.java
package com.enonic.wem.api.query.expr;

public abstract class OrderExpr
    implements Expression
{
    public enum Direction
    {
        ASC,
        DESC
    }

    private final Direction direction;

    public OrderExpr( final Direction direction )
    {
        this.direction = direction;
    }

    public final Direction getDirection()
    {
        return this.direction;
    }

    @Override
    public abstract String toString();

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( o == null || getClass() != o.getClass() )
        {
            return false;
        }

        final OrderExpr orderExpr = (OrderExpr) o;

        if ( direction != orderExpr.direction )
        {
            return false;
        }

        return true;
    }

    @Override
    public int hashCode()
    {
        return direction != null ? direction.hashCode() : 0;
    }
}
